src/module-4/side-bar/index.js

```diff
@@ -1,10 +1,105 @@
-import FiltersList from '../filters-list/index.js';
-
 export default class SideBar {
-  constructor (categoriesFilter = [], brandFilter = []) {
+  element;
+  subElements = {};
+  onCheck = () => {
+    const filters = this.getFilterCards();
+    Object.values(filters).forEach(item => item.reset());
+    this.update();
+    this.element.dispatchEvent(new CustomEvent('clear-filters'));
+  }
+
+  constructor (categoriesFilter = [], brandFilter = [], Component = {}) {
+    this.Component = Component;
     this.categoriesFilter = categoriesFilter;
     this.brandFilter = brandFilter;
+    this.render();
+    this.getSubElements();
+    this.addEventListeners();
+  }
+
+  get template() {
+    return `<div class="filters">
+  <div class="caption">
+    <span>Filters</span>
+    <button><img src="../../icons/arrows.png" alt="arrows"></button>
+  </div>
+
+  <div data-element="body" class="filter-cards">
+    ${this.getFilterCards().categoryFilterList.element.innerHTML}
+    <div class="divider"></div>
+    ${this.getFilterCards().brandFilterList.element.innerHTML}
+  </div>
+  <button data-element="button" id="reset">clear all filters</button>
+</div>`
+  }
+
+  render() {
+    const wrapper = document.createElement('div');
+    wrapper.innerHTML = this.template;
+    this.element = wrapper.firstChild;
+  }
+
+  remove() {
+    if(this.element) {
+      this.removeEventListeners();
+      this.element.remove();
+    }
+  }
+
+  destroy() {
+    this.remove();
+    this.element = null;
+    this.subElements = {};
+  }
+
+  reset() {
+    this.onCheck();
+  }
+
+  update() {
+    const filterElements = this.getFilterCardsTemplate();
+
+    this.subElements.body.replaceChildren(...filterElements);
+  }
+
+  getSubElements() {
+    const result = {};
+    const elements = this.element.querySelectorAll('[data-element]');
+    for (const subElement of elements) {
+      const name = subElement.dataset.element;
+      result[name] = subElement;
+    }
+    this.subElements = result;
+
+  }
 
-    // ... your logic
+  getFilterCards() {
+    const categoryFilterList = new this.Component({
```

Comment:
minor: here you can create `const Filter = this.Component` and then use `const categoryFilter = new Filter(...)`